src/controllers/author.controller.ts

```diff
@@ -58,3 +62,189 @@ export const author_detail = async (
     next(err);
   }
 };
+
+// GET create author
+export const getAuthorCreateForm = (req: Request, res: Response) => {
+  res.render('author/author_form');
+};
+
+// POST create author
+export const postAuthorCreateForm = [
+  body("first_name")
+    .trim()
+    .isLength({ min: 1 })
+    .escape()
+    .withMessage("First name must be specified.")
+    .isAlphanumeric()
+    .withMessage("First name has non-alphanumeric characters."),
+  body("family_name")
+    .trim()
+    .isLength({ min: 1 })
+    .escape()
+    .withMessage("Family name must be specified.")
+    .isAlphanumeric()
+    .withMessage("Family name has non-alphanumeric characters."),
+  body("date_of_birth", "Invalid date of birth")
+    .optional({ checkFalsy: true })
+    .isISO8601()
+    .toDate(),
+  body("date_of_death", "Invalid date of death")
+    .optional({ checkFalsy: true })
+    .isISO8601()
+    .toDate(),
+
+  async (
+    req: Request,
+    res: Response,
+    next: NextFunction,
+  ) => {
+    try {
+      const errors = validationResult(req);
+
+      if (!errors.isEmpty()) {
+        res.render('author/author_form', {
+          errors: errors.array(),
+        });
+        return;
+      }
+
+      const { first_name, family_name, date_of_birth } = req.body;
+
+      const author = new Author();
+      author.firstName = first_name;
+      author.familyName = family_name;
+      author.dateOfBirth = date_of_birth;
+      author.name = `${first_name} ${family_name}`;
+
+      await authorRepository.save(author);
+
+      res.redirect('/authors');
+    } catch (error) {
+      next(error);
+    }
+  }
+];
+
+// GET delete author
+export const author_delete_get = async (req: Request, res: Response) => {
```

Comment:
Dùng camelcase stype khi đặt tên biến, function, ... nhé
```suggestion
export const authorDeleteGet = async (req: Request, res: Response) => {
```